tests/kuttl/tests/keystone_scale/04-assert.yaml

```diff
@@ -0,0 +1,65 @@
+#
+# Check for:
+#
+# - 0 KeystoneAPI CR
+# - 0 Pods for KeystoneAPI CR
+#
+
+apiVersion: keystone.openstack.org/v1beta1
+kind: KeystoneAPI
+metadata:
+  finalizers:
+  - KeystoneAPI
+  name: keystone
+  namespace: openstack
+spec:
+  adminProject: admin
+  adminRole: admin
+  adminUser: admin
+  containerImage: quay.io/tripleozedcentos9/openstack-keystone:current-tripleo
+  customServiceConfig: |
+    [DEFAULT]
+    debug = true
+  databaseInstance: openstack
+  databaseUser: keystone
+  debug:
+    bootstrap: false
+    dbSync: false
+    service: false
+  passwordSelectors:
+    admin: AdminPassword
+    database: KeystoneDatabasePassword
+  preserveJobs: false
+  region: regionOne
+  replicas: 0
+  resources:
+    requests:
+      cpu: "1.0"
+      memory: 500Mi
+  secret: osp-secret
+status:
+  apiEndpoint:
+    admin: http://keystone-admin-openstack.apps-crc.testing
+    internal: http://keystone-internal-openstack.apps-crc.testing
+    public: http://keystone-public-openstack.apps-crc.testing
+  databaseHostname: openstack
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: keystone
+spec:
+  replicas: 0
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+namespaced: True
+commands:
+  - script: |
+      set -x
+      available=$(oc get deployment -n openstack keystone -o yaml | grep "availableReplicas: 0")
+      if [ -z "$available"  ]; then
```

Comment:
as far as I understand this assert passes if `availableReplicas: 0` is not in the output. But this means it also passes if `availableReplicas: 1` is in the output. And what we want is that it passes if no `availableReplicas` is in the output. So you probably want to change `grep "availableReplicas: 0"` to `grep "availableReplicas"`.

//later

Or go with @stuggi's suggestion to use an error if there is any pods with `service: keystone ` label.

Reply:
You're right, thanks. I'm definitely checking the pods with an `errors`, it'll be cleaner

Reply:
Moved the check to the `04-errors.yaml` file in the last commit